app/routes/datasets/dataset.py

```diff
@@ -21,21 +21,23 @@
 )
 from ...routes import dataset_dependency
 from ...settings.globals import READER_USERNAME
+from ...utils.rw_api import get_rw_user
 
 router = APIRouter()
 
 
 async def get_owner(
     dataset: str = Depends(dataset_dependency), user: User = Depends(get_manager)
 ) -> User:
-    """Returns the User making the request as long as that user is an admin or
-    the owner of the dataset, otherwise raises a 401."""
+    """Retrieves the user object that owns the dataset, or ADMIN user, if that
+    user is the one making the request, otherwise raises a 401."""
```

Comment:
This is a little clunky sounding. Perhaps we could re-word as "Retrieves the user object of the one making the request if that user either owns the dataset or is an ADMIN, otherwise raises a 401"
What do you think?